Library/Homebrew/macos_version.rb

```diff
@@ -76,7 +81,7 @@ def <=>(other)
       super
     end
 
-    @comparison_cache[other] = result unless frozen?
+    @comparison_cache[other.to_s] = result unless frozen?
```

Comment:
I wonder if we're comparing frozen objects too often, which prevents caching here. Maybe we should make the comparison_cache belong to the class, instead of the instance?

Reply:
Yeh, feels similar to me: this `frozen?` check seems nice to be able to avoid if possible.